leshan-server-core/src/main/java/org/eclipse/leshan/server/queue/PresenceServiceImpl.java

```diff
@@ -47,54 +45,43 @@ public void removeListener(PresenceListener listener) {
     }
 
     @Override
-    public boolean isOnline(Registration registration) {
-        return Presence.ONLINE.equals(clientStatus.get(registration.getEndpoint()));
+    public boolean isClientSleeping(Registration registration) {
+        LwM2mQueue queue = clientStatusList.get(registration.getEndpoint());
+        return queue.isClientSleeping();
     }
 
-    /**
-     * Sets the client identified by the given registration in {@link Presence#OFFLINE} mode if the current state is
-     * {@link Presence#ONLINE}. If the state is changed successfully, then the corresponding
-     * {@link PresenceListener} are notified.
-     * 
-     * @param registration lwm2m client registration data
-     */
-    public void setOffline(Registration registration) {
-        if (registration.usesQueueMode()
-                && transitState(registration.getEndpoint(), Presence.ONLINE, Presence.OFFLINE)) {
+    @Override
+    public void setAwake(Registration registration) {
+        if (registration.usesQueueMode()) {
+            clientStatusList.get(registration.getEndpoint()).setAwake();

```

Comment:
queue could be null.

Reply:
How? When registration is created, the `LwM2mClientStateListener` creates a queue object inside the `PresenceServiceImpl` linked to the endpoint name, so it shouldn't be null... 

If you don't agree with that what would you do? Surround with try, catch or check first if it's not null and then call `setAwake()`?